libyul/AsmJsonConverter.cpp

```diff
@@ -62,12 +63,13 @@ Json AsmJsonConverter::operator()(Literal const& _node) const
 		break;
 	case LiteralKind::String:
 		ret["kind"] = "string";
-		ret["hexValue"] = util::toHex(util::asBytes(_node.value.str()));
+		ret["unlimited"] = _node.value.unlimited();
```

Comment:
The inference of the kind of literal should not be based on the length but rather on whether it is an argument of a builtin with kind == string and and entry in `literalArguments` based on dialect. So it is indeed a property that stems from the function definition and can be inferred with knowledge of the dialect.
That being said, the parsing bit is currently only for inline assembly so the `unlimited` may be safely ignored for now. After a discussion with @ekpyron it seemed like a good idea to keep it around but thinking about it again, I don't think it is necessary at all and agree that it can be removed from the output structure.

Reply:
Ok so I removed the output for now, it's easy enough to add it back if it ever becomes relevant I suppose.